infoschema/tables.go

```diff
@@ -1828,6 +1843,112 @@ func dataForServersInfo() ([][]types.Datum, error) {
 	return rows, nil
 }
 
+func dataForTiDBClusterInfo(ctx sessionctx.Context) ([][]types.Datum, error) {
+	// get TiDB servers info.
+	tidbNodes, err := infosync.GetAllServerInfo(context.Background())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	rows := make([][]types.Datum, 0, len(tidbNodes))
+	for _, node := range tidbNodes {
+		tp := "tidb"
+		name := fmt.Sprintf("%s-%d", tp, len(rows))
+		addr := fmt.Sprintf("%s:%d", node.IP, node.Port)
+		statusAddr := fmt.Sprintf("%s:%d", node.IP, node.StatusPort)
+		row := types.MakeDatums(
+			len(rows)+1,
+			tp,
+			name,
+			addr,
+			statusAddr,
+			node.Version,
+			node.GitHash,
+		)
+		rows = append(rows, row)
+	}
+
+	// get PD servers info.
+	store := ctx.GetStore()
+	etcd, ok := store.(tikv.EtcdBackend)
+	if !ok {
+		return nil, errors.Errorf("%T not an etcd backend", store)
+	}
+	for i, addr := range etcd.EtcdAddrs() {
+		addr = strings.TrimSpace(addr)
+
+		// get pd version
+		url := fmt.Sprintf("http://%s/pd/api/v1/config/cluster-version", addr)
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		pdVersion, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		terror.Log(resp.Body.Close())
```

Comment:
Why log?

Reply:
`terror.Log` it a utility function equals `if err != nil {...}`.